### ISSUE TITLE
User interface to GSL state updaters 

### ISSUE THREAD

**Issue by mstimberg** [brian-team/brian2#867]
To use the GSL state updater, you currently (i.e. in the @CharleeSF's GSL branch -- let's not focus on the question of how to ship it (#866) for now) use `NeuronGroup` (or `Synapses`, or `SpatialNeuron`) with `method="GSL_stateupdater"`. This will by default use a the `rkf45` method ("A good general-purpose integrator" according to the docs). To use a different method, you'll have to do:
```Python
 method='GSL_stateupdater', method_options={'integrator', 'rk2'}
```
(with `method_options` you can define other stuff as well, but let's ignore that)
I think this might be a bit too verbose and not very consistent with the other state updaters. My proposal -- given that there are only 5 algorithms that we can use at the moment (no support for providing the Jacobian, etc.) -- is to have 5 methods:
`method='GSL_rk2'`, `method='GSL_rk4'`, `method='GSL_rkf45'`, `method='GSL_rkck'`, `method='GSL_rk8pd'`
Or maybe this is too detailed and users will actually never switch between these methods? In that case, simply `method='GSL'` maybe?
@thesamovar : Thoughts?

**Comment by thesamovar** [comment]
I like your version. What about including a method='gsl' as an alias for gsl_rkf45? Also, the strings are case insensitive right? 

Dan

**Comment by mstimberg** [comment]
> I like your version. What about including a method='gsl' as an alias for gsl_rkf45?

Ah, good compromise, I like it.

> Also, the strings are case insensitive right?

Yes, that's a general thing we do for the state updater names (e.g. 'rk2' == 'RK2')

**Comment by mstimberg** [comment]
Has been implemented via #870.
